R/estimate_infections.R

```diff
@@ -131,6 +133,24 @@ estimate_infections <- function(reported_cases,
                                 weigh_delay_priors = TRUE,
                                 id = "estimate_infections",
                                 verbose = interactive()) {
+  # Validate inputs
+  check_reports_valid(reported_cases, for_estimate_secondary = FALSE)
+  assert_class(generation_time, "generation_time_opts", null.ok = TRUE)
+  assert_class(delays, "delay_opts", null.ok = TRUE)
+  assert_class(truncation, "trunc_opts", null.ok = TRUE)
+  assert_class(rt, "rt_opts", null.ok = TRUE)
+  assert_class(backcalc, "backcalc_opts", null.ok = TRUE)
+  assert_class(gp, "gp_opts", null.ok = TRUE)
+  assert_class(obs, "obs_opts", null.ok = TRUE)
+  assert_class(stan, "stan_opts", null.ok = TRUE)
```

Comment:
I might be wrong (and it would be easy to test) but I don't think any of these except `rt` should have `null.ok = TRUE` - where setting `rt = NULL` perhaps somewhat confusingly indicates to not use a renewal model (which will eventually be addressed by #345).

Reply:
Yes, you're right. `rt` and `gp` can be null but the others can't be.